### PR TITLE
RavenDB-15912 on data move should forget about cached compressed ids

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Voron/Data/RawData/ActiveRawDataSmallSection.cs [hunk 1]
```diff
@@ -176,11 +177,11 @@ public string DebugDump(RawDataSmallPageHeader* pageHeader)
                         if (oldSize->IsCompressed)
                         {
                             using var __ = Table.DecompressValue(_transaction, entryPos, size, out var buffer);
-                            OnDataMoved(prevId, newId, buffer.Ptr, buffer.Length);
+                            OnDataMoved(prevId, newId, buffer.Ptr, buffer.Length, compressed: true);
```
Comment (ayende) [inline src/Voron/Data/RawData/ActiveRawDataSmallSection.cs:180]: Why not call this here?
` _tx.CachedDecompressedBuffersByStorageId.Remove(previousId);`

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Voron/Data/RawData/ActiveRawDataSmallSection.cs (hunk 1 above)
Comment (karmeli87) [reply src/Voron/Data/RawData/ActiveRawDataSmallSection.cs:180]: this is only one place.
You have this one when the page is defragged and you have when the density is lower than 0.15 we merge sections.

Comment (karmeli87) [reply src/Voron/Data/RawData/ActiveRawDataSmallSection.cs:180]: both which call `OnDataMoved`

**Review by ayende — APPROVED** [review]

File: src/Voron/Data/RawData/ActiveRawDataSmallSection.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Voron/Data/RawData/ActiveRawDataSmallSection.cs:180]: I like Karmel's approach. It's less error prone.

**Review by arekpalinski — APPROVED** [review]

**Pull request merged**